flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SingleLogicalSlot.java

```diff
@@ -168,6 +185,11 @@ public void release(Throwable cause) {
 		releaseFuture.complete(null);
 	}
 
+	@Override
+	public boolean willOccupySlotIndefinitely() {
```

Comment:
It is intentional.
SingleLogicalSlot has 2 roles. One is logical slot and the other is physical slot payload.
It will occupy a physical slot indefinitely iff it will be occupied indefinitely by a task.